packages/evm/src/eof/container.ts

```diff
@@ -272,11 +272,10 @@ class EOFHeader {
 
   // Returns the code section for a given program counter position
   getSectionFromProgramCounter(programCounter: number) {
-    if (
-      programCounter < 0 ||
-      programCounter >
-        this.codeStartPos[this.codeStartPos.lastIndex] + this.codeSizes[this.codeSizes.lastIndex]
-    ) {
+    const lastIndex = this.codeSizes.length - 1
+    const lastStart = this.getCodePosition(lastIndex)
+    const lastEnd = lastStart + this.codeSizes[lastIndex]
+    if (programCounter < 0 || programCounter > lastEnd) {
```

Comment:
Please always give some context on these kind of fixes and not magically provide solutions on these kind of things. So what happened here? How could this just break from one PR to the other or from whereever? And how what is the fix doing?

Will nevertheless take this in, since this only touches somewhat legacy EOF code. Nevertheless, small additional explanation would be good!